core/src/main/scala/org/http4s/Message.scala

```diff
@@ -77,35 +77,9 @@ sealed trait Message[F[_]] extends Media[F] { self =>
     * @return a new message with the new body
     */
   def withEntity[T](b: T)(implicit w: EntityEncoder[F, T]): Self = {
-    val entity = w.toEntity(b)
-    val hs = entity.length match {
-      case Some(l) =>
-        `Content-Length`
-          .fromLong(l)
-          .fold[Headers](
-            _ => {
-              Message.logger.warn(s"Attempt to provide a negative content length of $l")
-              w.headers
-            },
-            cl => Headers(cl :: w.headers.toList))
-      case None => w.headers
-    }
-    change(body = entity.body, headers = headers ++ hs)
+    change(entity = w.toEntity(b))
   }
 
-  /** Sets the entity body without affecting headers such as `Transfer-Encoding`
-    * or `Content-Length`. Most use cases are better served by [[withEntity]],
-    * which uses an [[EntityEncoder]] to maintain the headers.
-    */
-  def withBodyStream(body: EntityBody[F]): Self =
```

Comment:
We might consider a deprecation cycle on this one to help people migrate.

Reply:
The difficulty with it is what the scala doc says. There is now no way to not effect the headers like that.